lib/compile/formats.js

```diff
@@ -33,7 +33,8 @@ formats.fast = {
   // email (sources from jsen validator):
   // http://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address#answer-8829363
   // http://www.w3.org/TR/html5/forms.html#valid-e-mail-address (search for 'willful violation')
```

Comment:
The more I read into this, the more confused I get.
"I am an email"@email.com should be valid apparently, and you an IP in the domain using square brackets should also be allowed me@[127.0.0.1] (although wikipedia references some article that mentions this will trigger quite a lot of spam filters, so in practice people probably shouldn't be doing this)

Regarding underscores: most registrars don't allow you to register something with an underscore in the root domain I've found, but eg AWS Route53 does allow you to put underscores in A records and CNAME records for subdomains (and applications can work with these without an issue)

(I'll probably be a bit slower over the next days to reply, things are rather busy at the moment)